<launch>

<!-- Camera -->
    <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen" >
        <param name="video_device" value="/dev/video0" />
	    	<param name="image_width" value="400" />
    		<param name="image_height" value="380" />
    		<param name="pixel_format" value="yuyv" />
    		<param name="camera_frame_id" value="usb_cam" />
    		<param name="io_method" value="mmap"/>
        <param name="autofocus" value="false"/>
        <param name="camera_info_url" value="file:///$(find endoscopy_demo)/config/cameras/endoscope_v1001_cam_calib.yaml"/>
   </node>


  <group ns="usb_cam">
      <node pkg="image_proc" type="image_proc" name="image_proc" />
  </group>

<!-- Optical flow  -->
    <node name="flow" pkg="opencv_apps" type="phase_corr" >
    <remap from="image" to="/usb_cam/image_raw" />
    <param name="use_camera_info" value="true" />
    <param name="debug_view" value="true" />
    <param name="block_size" value="10" />

  </node>
  
<!-- Navigation node  -->
  <node name="pov_nav" pkg="pov_mag_navigation" type="pov_navigation_node.py" output="screen"  >
      <remap from="/desired_field" to="/backward_cmag/field"/>
    </node>

<!-- PS navi -->
  <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
      <param name="autorepeat_rate" value="30"/>
    </node>
  <node name="ps_navi_controller" pkg="pov_mag_navigation" type="ps_navi_controller" output="screen" >
  </node>

<!-- Dynamic reconfigure -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" >
    <rosparam command="load" file="$(find pov_mag_navigation)/param/parameters_pov_transnasal.yaml" />
  </node>


<!-- Rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find pov_mag_navigation)/rviz/pov_nav_cmag.rviz" />

<!-- eMNS -->
   <node pkg="cardiomag_controller" type="cardiomag_controller" name="cardiomag_controller" respawn="false">
        <param name="port" type="int" value="3072" />
        <remap from="/cmag/desired_currents" to="/desired_currents"/>
        <param name="use_advancer" type="bool" value="false" />
    </node>

    <group> 
        <remap from="/mag_calculator/mag_field" to="/desired_field"/>
        <include file="$(find mag_calculator)/launch/backward_cmag.launch"/>
    </group>

    <node pkg="tf" type="static_transform_publisher" name="mns_broadcaster" args="-0.00851678 0.03142507 0.03213744 5.92061853e-03 8.29798521e-04 5.10459394e-03 9.99969100e-01 world mns 100"/>


</launch>